syntax = "proto3";

import "google/protobuf/empty.proto";

package primitives.int;

service IntService {
  rpc Set(IntSetRequest) returns (google.protobuf.Empty);
  rpc Get(IntGetRequest) returns (IntGetResponse);
  rpc Delete(IntDeleteRequest) returns (IntDeleteResponse);
}

message IntSetRequest {
  string id = 1;
  int64 value = 2;
  optional uint64 ttl_seconds = 3;
}

message IntGetRequest {
  string id = 1;
}

message IntGetResponse {
  int64 value = 1;
}

message IntDeleteRequest {
  string id = 1;
}

message IntDeleteResponse {
  int64 removed_value = 1;
}
